examples/twitter-compose-with-typeahead/src/Autocomplete.tsx

```diff
@@ -0,0 +1,259 @@
+    getSources({ query }) {
+      const cursorPosition = inputRef.current?.selectionEnd || 0;
+      const activeToken = getActiveToken(query, cursorPosition);
+
+      if (activeToken?.word && isValidTwitterUsername(activeToken?.word)) {
+        return [
+          {
+            sourceId: 'accounts',
+            onSelect({ item, setQuery }) {
+              const [index] = activeToken.range;
+              const replacement = `@${item.handle}`;
+              const newQuery = replaceAt(
+                query,
+                replacement,
+                index,
+                activeToken.word.length
+              );
+
+              setQuery(newQuery);
+
+              if (inputRef.current) {
+                inputRef.current.selectionEnd = index + replacement.length;
+              }
+            },
+            getItems() {
+              return getAlgoliaResults({
+                searchClient,
+                queries: [
+                  {
+                    indexName: 'autocomplete_twitter_accounts',
+                    query: activeToken.word.slice(1),
+                    params: {
+                      hitsPerPage: 8,
+                    },
+                  },
+                ],
+              });
+            },
+          },
+        ];
+      }
+
+      return [];
+    },
+  });
+
+  function onInputNavigate() {
+    const cursorPosition = inputRef.current?.selectionEnd || 0;
+    const activeToken = getActiveToken(state.query, cursorPosition);
+    const shouldOpen = isValidTwitterUsername(activeToken?.word || '');
+
+    autocomplete.setIsOpen(shouldOpen);
+    autocomplete.refresh();
+  }
+
+  const { top, height } = inputRef.current
+    ? getCaretCoordinates(inputRef.current, inputRef.current?.selectionEnd)
+    : { top: 0, height: 0 };
+
+  const inputProps = autocomplete.getInputProps({
+    inputElement: (inputRef.current as unknown) as HTMLInputElement,
+  });
+
+  return (
+    <div {...autocomplete.getRootProps({})}>
+      <div className="box">
+        <div className="box-body">
+          <div className="box-avatar">
+            <img src="https://robohash.org/autocomplete" alt="You" />
+          </div>
+          <div className="box-compose">
+            <form
+              {...autocomplete.getFormProps({
+                inputElement: (inputRef.current as unknown) as HTMLInputElement,
+              })}
+            >
+              <textarea
+                className="box-textbox"
+                ref={inputRef}
+                {...inputProps}
+                spellCheck={false}
+                autoFocus={true}
+                maxLength={280}
+                onKeyUp={(event) => {
+                  if (['ArrowLeft', 'ArrowRight'].includes(event.key)) {
+                    onInputNavigate();
+                  }
+                }}
+                onClick={(event) => {
+                  inputProps.onClick(event);
+
+                  onInputNavigate();
+                }}
```

Comment:
You can pass these properties to the `getInputProps` call so that you only need to spread `inputProps` then. (Providing `maxLength` to `getInputProps` does some additional work that really guards against sending longer queries.)

Reply:
Done in fd3acbbc.